mats/6.ms

```diff
@@ -3409,6 +3410,20 @@
   (equal? (path-extension "c:..") "")
   (equal? (path-extension "c:") "")
 
+  (equal? (path-build "" "b.ss") "b.ss")
+  (if (windows?)
+      (equal? (path-build "." "b.ss") ".\\b.ss")
+      (equal? (path-build "." "b.ss") "./b.ss"))
+  (if (windows?)
+      (equal? (path-build ".\\" "b.ss") ".\\b.ss")
+      (equal? (path-build "./" "b.ss") "./b.ss"))
+  (if (windows?)
+      (equal? (path-build "foo" "b.ss") "foo\\b.ss")
+      (equal? (path-build "foo" "b.ss") "foo/b.ss"))
+  (if (windows?)
+      (equal? (path-build "quux\\" "b.ss") "quux\\b.ss")
+      (equal? (path-build "quux/" "b.ss") "quux/b.ss"))
+
```

Comment:
Simplify:
```
(equal? (path-build "" "fn") "fn")
(equal? (path-build "dir" "fn") (format "dir~cfn" (directory-separator)))
(equal? (path-build (format "dir~c" (directory-separator)) "fn") (format "dir~cfn" (directory-separator)))
```

Reply:
I thought of doing that, but in a sense, I find it a bit less readable. Anyway, just did what you suggested.